frappe/core/doctype/prepared_report/prepared_report.py

```diff
@@ -175,6 +178,29 @@ def make_prepared_report(report_name, filters=None):
 	return {"name": prepared_report.name}
 
 
+@frappe.whitelist()
+def stop_prepared_report(report_name: str):
+	"""Stop a running Prepared Report job."""
+	prepared_report = frappe.get_doc("Prepared Report", report_name)
+	prepared_report.check_permission("write")
+
+	job_id = prepared_report.job_id
+	if not job_id.startswith(frappe.local.site):
+		frappe.throw(f"Invalid job_id: must start with {frappe.local.site}")
+
+	try:
+		send_stop_job_command(connection=get_redis_conn(), job_id=job_id)
+		frappe.db.set_value(
+			"Prepared Report",
+			prepared_report.name,
+			{"status": "Cancelled"},
+		)
+		frappe.db.commit()
```

Comment:
Commit isn't necessary for `POST` requests. 

https://docs.frappe.io/framework/user/en/api/database#database-transaction-model